### DISCUSSION TITLE
Is there a shortcode to get the relative path of the current file?

### DISCUSSION THREAD

**Discussion by jlperla** [quarto-dev/quarto-cli#11719] [Q&A]
### Description

I have a file, e.g. in `slides/intro.qmd` which I am using to buid revealjs slides (which end up as `slides/intro.html` in my render).
I am following advice to also add in a decktape post-render script to generate a `slides/intro.pdf` in my toolchain.

My question is how I can, generally, refer to a PDF created for a particular file?  Is there a way with a shortcode to get the `slides/intro` in some way, which I can then copy/paste into my templates to work for linking to any files?

**Comment by cscheid** [comment]
```lua
function Pandoc(doc)
  print(quarto.doc.input_file)
  print(quarto.doc.output_file)
end
```

Embarrassingly, we [don't appear to document this](https://quarto.org/docs/extensions/lua-api.html). We should, it's a public API and we even have lua-types declarations for it. I should also note that for `format: pdf`, you will get a `.tex` file instead of a `.pdf` file (and a `.typ` file for `format: typst`).

For pre- and post-render scripts, you have (documented, phew!) [environment variables](https://quarto.org/docs/projects/scripts.html#pre-and-post-render)

**Reply by cderv** [reply]
> Embarrassingly, we [don't appear to document this](https://quarto.org/docs/extensions/lua-api.html).

Still an open documentation issue 
- https://github.com/quarto-dev/quarto-cli/issues/4499

Also this comes from the exact same use case 
- https://github.com/quarto-dev/quarto-cli/discussions/2230 leading to https://github.com/quarto-dev/quarto-cli/issues/2249 which was closed prematurly due to discussion drift in thread

There is also still the idea to expose Lua Variable from pandoc and quarto as a shortcode 
- https://github.com/quarto-dev/quarto-cli/issues/4770

And extension allows to do that right now through filter and shortcode: https://github.com/mcanouil/quarto-lua-env

````
{{< lua-env quarto.doc.input_file >}}
````

And this is for the story around this topic 😄

**Comment by jlperla** [comment]
Thanks! How would I add this to my file to get it automatically, swap out the html to pdf, and embed with {{}}?
